text/0036-auto-instrumentation-ui-thread.md

```diff
@@ -0,0 +1,127 @@
+* Start Date: 2022-11-08
+* RFC Type: decision
+* RFC PR: [#36](https://github.com/getsentry/rfcs/pull/36)
+* RFC Status: -
+* RFC Driver: [Roman Zavarnitsyn](https://github.com/romtsn)
+
+# Summary
+
+Add new `blocked_ui_thread` attribute to the `data` key-value map of the [`Span`](https://develop.sentry.dev/sdk/event-payloads/span/) interface, indicating whether
+an auto-instrumented span has run its entire duration on the UI/Main thread.
```

Comment:
I read somewhere that ui thread blocked thresholds are:

> performance interaction latency thresholds
> good 0-199 ms
> needs improvements 200 - 499 ms
> poor >= 500  ms

So in case the total amount of all the blocked UI thread spans, we could actually raise performance issues in case it's `poor`, maybe?

Reply:
Probably here https://web.dev/inp/#whats-a-good-inp-value

idk, on the one hand android's strictmode warns you about any kind of file i/o performed on main thread, on the other hand it's probably too strict. For now we've started with 16ms (that's when a frame drop occurs on most devices), but I guess we'll tune this based on customer's data/feedback.